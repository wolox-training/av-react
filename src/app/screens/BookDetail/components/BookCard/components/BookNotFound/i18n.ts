import i18next from 'i18next';

i18next.addResources('es', 'BookNotFound', {
  notFound: 'Libro no encontrado o sin informaci√≥n'
});
