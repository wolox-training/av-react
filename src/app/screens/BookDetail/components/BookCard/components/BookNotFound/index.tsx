import i18next from 'i18next';
import React from 'react';

import styles from './styles.module.scss';

function BookNotFound() {
  return (
    <div className={`${styles.wrapper} row center`}>
      <span className={styles.notFound}>{i18next.t('BookNotFound:notFound')}</span>
    </div>
  );
}

export default BookNotFound;
