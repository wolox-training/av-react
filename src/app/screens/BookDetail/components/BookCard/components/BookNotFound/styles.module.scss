.wrapper {
  margin-top: 20px;

  .not-found {
    font-size: 32px;
    font-weight: 700;
  }
}
