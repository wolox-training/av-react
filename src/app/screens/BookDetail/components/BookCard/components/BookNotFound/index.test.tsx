import React from 'react';
import { render, screen } from '@testing-library/react';

import BookNotFound from './';

describe('BookNotFound test', () => {
  let container: Element | null = null;

  beforeEach(() => {
    const { container: renderContainer } = render(<BookNotFound />);
    container = renderContainer;
  });

  test('Should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  test('Should render not found text', () => {
    expect(screen.getByText('BookNotFound:notFound')).toBeInTheDocument();
  });
});
