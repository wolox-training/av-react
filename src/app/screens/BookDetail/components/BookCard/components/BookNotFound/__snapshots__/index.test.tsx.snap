// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookNotFound test Should match snapshot 1`] = `
<div>
  <div
    class="wrapper row center"
  >
    <span
      class="notFound"
    >
      BookNotFound:notFound
    </span>
  </div>
</div>
`;
